database/src/test/scala/za/co/absa/atum/database/runs/GetPartitioningAncestorsIntegrationTests.scala

```diff
@@ -0,0 +1,391 @@
+        val row = queryResult.next()
+        row.getLong("id_partitioning").get
+      }
+
+    val partitioningID8 = function(createPartitioningFn)
+      .setParam("i_partitioning", partitioning8)
+      .setParam("i_by_user", "Grandson")
+      .setParam("i_parent_partitioning", partitioning5)
+      .execute { queryResult =>
+        val row = queryResult.next()
+        row.getLong("id_partitioning").get
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partitioningID7)
+      .setParam("i_fk_partitioning", partitioningID8)
+      .setParam("i_by_user", "Grandson")
+      .execute { queryResult =>
+        queryResult.next()
+      }
+
+    //Used Linked Hash Map to keep structure and order
+    val resultsMap = mutable.LinkedHashMap(
+      "Grandma" -> (partitioningID1, expectedPartitioning1),
+      "Grandpa" -> (partitioningID2, expectedPartitioning2),
+      "Mother" -> (partitioningID3, expectedPartitioning3),
+      "Father" -> (partitioningID4, expectedPartitioning4),
+      "Son" -> (partitioningID5, expectedPartitioning5),
+      "Daughter" -> (partitioningID6, expectedPartitioning6),
+      "Granddaughter" -> (partitioningID7, expectedPartitioning7)
+    )
+    //Data Preparation Step End --------------------------------------------------------------------------------------
+
+    //Test 1 Ancestor
+    function(getAncestorsFn)
+      .setParam("i_id_partitioning", partitioningID3)
+      .execute { queryResult =>
+        val row = queryResult.next()
+        val returnedPartitioning = row.getJsonB("partitioning").get
+        val returnedPartitioningParsed = parse(returnedPartitioning.value)
+          .getOrElse(fail("Failed to parse returned partitioning"))
+        assert(row.getInt("status").contains(10))
+        assert(row.getString("status_text").contains("OK"))
+        assert(row.getLong("ancestor_id").contains(resultsMap("Grandma")._1))
+        assert(returnedPartitioningParsed == resultsMap("Grandma")._2)
+        assert(row.getString("author").contains("Grandma"))
+        assert(!queryResult.hasNext)
+      }
+
+    //Test Multiple Ancestors
+    function(getAncestorsFn)
+      .setParam("i_id_partitioning", partitioningID5)
+      .execute { queryResult =>
+        var row = queryResult.next()
+        var returnedPartitioning = row.getJsonB("partitioning").get
+        var returnedPartitioningParsed = parse(returnedPartitioning.value)
+          .getOrElse(fail("Failed to parse returned partitioning"))
+        //Used breakable to be able to break the loop
+        breakable {
+          for ((k, v) <- resultsMap) {
+            assert(row.getInt("status").contains(10))
+            assert(row.getString("status_text").contains("OK"))
+            assert(row.getLong("ancestor_id").contains(v._1))
+            assert(returnedPartitioningParsed == v._2)
+            assert(row.getString("author").contains(k))
+            if (!queryResult.hasNext) break()
```

Comment:
this is a very non-functional, procedural-like code, which is, no matter the style, very non-deterministic. I would prepare exact expected parents and loop over them, not like this